src/Sparrow/Utils/TimeoutManager.cs

```diff
@@ -64,6 +64,19 @@ public void Dispose()
 
         private static async Task WaitForInternal(TimeSpan time, CancellationToken token)
         {
+            if (time == Timeout.InfiniteTimeSpan)
+            {
+                if (token == CancellationToken.None || token.CanBeCanceled == false)
+                {
+                    await InfiniteTask;
+                }
+                else
+                {
+                    await new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously).Task;
```

Comment:
It is. We connect it to the cancellation token in the method that call this method.